peak/mapper/mapper.py

```diff
@@ -4,145 +4,134 @@
-
-    logging.debug("Enumerating bv instructions")
-    bv_isa_list = list(filter(f_fun, bv_isa.enumerate()))
-    bv_isa_len = len(bv_isa_list)
-    logging.debug("Enumerating smt instructions")
-    smt_isa_list = list(filter(f_fun, smt_isa.enumerate()))
-    smt_isa_len = len(smt_isa_list)
-
-    logging.debug("Starting search")
-
-    for ii,smt_inst in enumerate(smt_isa_list):
-        logging.debug(f"inst {ii+1}/{bv_isa_len}")
-        logging.debug(smt_inst)
-
-        for bi,binding in enumerate(bindings):
-            binding_dict = {k : core_smt_vars[v] if v is not None else peak_inputs[k](0) for v,k in binding}
-            name_binding = {k : v if v is not None else 0 for v,k in binding}
-
-            rvals = peak_inst(smt_inst, **binding_dict)
-            if not isinstance(rvals, tuple):
-                rvals = rvals,
-
-            for idx, bv in enumerate(rvals):
-                if isinstance(bv, (SMTBit, SMTBitVector)) and bv.value.get_type() == core_smt_expr.value.get_type():
-                    with smt.Solver(solver_name, logic=QF_BV) as solver:
-                        if check_equal(solver_name, binding_dict, bv, core_smt_expr, name_binding):
-                            #Create output and input map
-                            output_map = {"out":list(peak_class.__call__._peak_outputs_.items())[idx][0]}
-                            input_map = {}
-                            for k,v in name_binding.items():
-                                if v == 0:
-                                    v = "0"
-                                elif v == "in_":
-                                    v = "in"
-                                input_map[v] = k
-
-                            mapping = dict(
-                                instruction=bv_isa_list[ii],
-                                output_map=output_map,
-                                input_map=input_map
-                            )
-                            yield mapping
-                            found  += 1
-                            if found >= max_mappings:
-                                return
-
-def check_equal(solver_name, smt_vars, expr1, expr2, name_binding):
-    with smt.Solver(solver_name, logic=QF_BV) as solver:
-        expr = expr1 != expr2
-        solver.add_assertion(expr.value)
-        if not solver.solve():
-            return True
-        else:
-            model = solver.get_model()
-            model = {k : model[v.value] for k,v in smt_vars.items()}
-            logging.log(logging.DEBUG - 1, name_binding)
-            logging.log(logging.DEBUG - 1, model)
-            return False
+        custom_enumeration : tp.Mapping[type,tp.Callable] = {}
+    ):
+        self.solver_name = solver_name
+        self.custom_enumeration = custom_enumeration
+
+        arch_smt = arch_fclosure(SMTBitVector.get_family())
+        self.arch_sim = arch_smt()
+
+        self.arch_inputs = arch_smt.__call__._peak_inputs_
```

Comment:
Should add a method to `Peak` to get this information

Reply:
Added separate issue (as other places use this)

Reply:
gotcha